### PR TITLE
Adding AES encryption service and unit tests.

### LINKED ISSUE #4: Add AesEncryptionService Functionality
A new implementation should be added to the library that supports string encryption & decryption using a shared key value.  This should be implemented supporting an options configuration to inject the key & iv values, and a simple interface for usage

*Options*
```
public class AesEncryptionServiceOptions
{
    public string Key {get; set;}
    public string IV {get; set;}
}
```
*API*
```
public interface IAesEncryptionService
{
    string Encrypt(string plainTextInput);
    string Decrypt(string encryptedInput);
}
```

### PR DESCRIPTION
**Pull request by kenny-sellers**
Closes #4

### CODE REVIEW

File: src/NetCore.Utilities.Tests/AesEncryptionServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        {
+            //Act
+            var encrypted = _service.Encrypt(input);
+            var decrypted = _service.Decrypt(encrypted);
+
+            //Assert
+            Assert.Equal(input, decrypted);
+        }
+
+        [Theory]
```
Comment (mitchelsellers) [inline src/NetCore.Utilities.Tests/AesEncryptionServiceTests.cs:39]:
```suggestion
        ///<summary>
        ///    This test is need to ensure that stored values, that were encrypted with the known IV/Secret still can be properly decrypted.  This was necessary after weird behaviors noticed in .NET 5 -> .NET 6 transition
        /// </summary>
        [Theory]
```

File: src/NetCore.Utilities/AesEncryptionService.cs [hunk 2]
```diff
@@ -0,0 +1,134 @@
+﻿using Microsoft.Extensions.Options;
+using System;
+using System.IO;
+using System.Security.Cryptography;
+
+namespace ICG.NetCore.Utilities
+{
+    /// <summary>
+    ///     Represents a service that can encrypt and decrypt strings using symmetric AES encryption.
```
Comment (mitchelsellers) [inline src/NetCore.Utilities/AesEncryptionService.cs]:
```suggestion
    ///     Represents a service that can encrypt and decrypt strings using symmetric AES encryption, using a stored key/iv.
```

File: src/NetCore.Utilities/AesEncryptionService.cs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        /// </summary>
+        /// <param name="plainTextInput">The plain text string to be encrypted</param>
+        /// <exception cref="ArgumentNullException">If [plainTextInput] is null</exception>
+        /// <exception cref="ArgumentNullException">If option key is null</exception>
+        /// <exception cref="ArgumentNullException">If option IV is null</exception>
+        /// <returns>A encrypted string representing the provided plain text string.</returns>
+        string Encrypt(string plainTextInput);
+
+        /// <summary>
+        ///     Decrypts the provided string into an AES encrypted string.
```
Comment (mitchelsellers) [inline src/NetCore.Utilities/AesEncryptionService.cs]:
```suggestion
        ///     Decrypts the provided string from an AES encrypted string back to plain text.
```

File: src/NetCore.Utilities/AesEncryptionService.cs [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+namespace ICG.NetCore.Utilities
+{
+    /// <summary>
+    ///     Represents a service that can encrypt and decrypt strings using symmetric AES encryption.
+    /// </summary>
+    public interface IAesEncryptionService
+    {
+        /// <summary>
+        ///     Encrypts the provided string into an AES encrypted string.
```
Comment (mitchelsellers) [inline src/NetCore.Utilities/AesEncryptionService.cs]:
```suggestion
        ///     Encrypts the provided plain-text string into an AES encrypted string.
```

File: src/NetCore.Utilities/AesEncryptionService.cs [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+        public string Decrypt(string encryptedInput)
+        {
+            // Check arguments.
+            if (string.IsNullOrEmpty(encryptedInput))
+                throw new ArgumentNullException("encryptedInput");
+            if (string.IsNullOrEmpty(_serviceOptions.Key))
+                throw new ArgumentNullException("Key");
+            if (string.IsNullOrEmpty(_serviceOptions.IV))
+                throw new ArgumentNullException("IV");
+            byte[] encrypted;
```
Comment (mitchelsellers) [inline src/NetCore.Utilities/AesEncryptionService.cs]: This is not needed and should be removed

File: src/NetCore.Utilities/DependencyResolution/StartupExtensions.cs [hunk 6]
```diff
@@ -11,7 +12,7 @@ public static class StartupExtensions
         /// Registers the items included in the ICG NetCore Utilities project for Dependency Injection
         /// </summary>
         /// <param name="services">Your existing services collection</param>
```
Comment (mitchelsellers) [inline src/NetCore.Utilities/DependencyResolution/StartupExtensions.cs:14]: Need to add <param> tag for configuration

**Review by mitchelsellers — CHANGES_REQUESTED** [review]
Overall this looks good, just a few comments on documentation changes and a few cleanup items.

Additionally, we should add information to the Readme.md file to document usage of these options.

Lastly, we might want to consider an additional enhancement for this where the Secret/IV is passed in, we can make a separate PR for this.

**NOTE** This PR changes a public API interface and is a breaking change, this dependency change will impact other libraries.

File: src/NetCore.Utilities/AesEncryptionService.cs (hunk 5 above)
Comment (mitchelsellers) [reply src/NetCore.Utilities/AesEncryptionService.cs]: This is ONLY talking about the `byte[] encrypted;` line

**Comment by kenny-sellers** [comment]
@mitchelsellers Should be ready for your review

**Review by mitchelsellers — APPROVED** [review]

**Pull request merged**